hubspot_sync/api.py

```diff
@@ -0,0 +1,482 @@
+        "leadership_level": "leadership_level",
+        "highest_education": "highest_education",
+    }
+
+    user = User.objects.get(id=user_id)
+    if not hasattr(user, "profile"):
+        return [{}]
+    properties = UserSerializer(user).data
+    properties.update(properties.pop("legal_address") or {})
+    properties.update(properties.pop("profile") or {})
+    properties["work_experience"] = properties.pop("years_experience", None)
+    if "street_address" in properties:
+        properties["street_address"] = "\n".join(properties.pop("street_address"))
+    hubspot_props = transform_object_properties(properties, contact_properties_map)
+    return make_object_properties_message(hubspot_props)
+
+
+def make_product_sync_message(bootcamp_run_id):
+    """
+    Create the body of a sync message for a product.
+
+    Args:
+        bootcamp_run_id (int): Bootcamp run id
+
+    Returns:
+        list: dict containing serializable sync-message data
+    """
+    bootcamp_run = BootcampRun.objects.get(id=bootcamp_run_id)
+    properties = HubspotProductSerializer(instance=bootcamp_run).data
+    return make_object_properties_message(properties)
+
+
+def make_deal_sync_message(application_id):
+    """
+    Create the body of a sync message for a deal.
+
+    Args:
+        application_id (int): BootcampApplication id
+
+    Returns:
+        list: dict containing serializable sync-message data
+    """
+    application = BootcampApplication.objects.get(id=application_id)
+    properties = HubspotDealSerializer(instance=application).data
+    return make_object_properties_message(properties)
+
+
+def make_line_sync_message(application_line_id):
+    """
+    Create the body of a sync message for a Line Item.
+
+    Args:
+        application_line_id (int):BootcampApplicationLine id
+
+    Returns:
+        list: dict containing serializable sync-message data
+    """
+    line = BootcampApplicationLine.objects.get(id=application_line_id)
+    properties = HubspotLineSerializer(instance=line).data
+    properties["quantity"] = 1
+    return make_object_properties_message(properties)
+
+
+def get_hubspot_id_for_object(
+    obj: BootcampApplication or BootcampApplicationLine or BootcampRun or User,
+    raise_error: bool = False,
+):
+    """
+    Get the hubspot id for an object, querying Hubspot if necessary
+
+    Args:
+        obj(object): The object (BootcampApplication or BootcampRun or User) to get the id for
+        raise_error(bool): raise an error if not found (default False)
+
+    Returns:
+        The hubspot id for the object if it has been previously synced to Hubspot.
+        Raises a ValueError if no matching Hubspot object can be found.
+    """
+
+    content_type = ContentType.objects.get_for_model(obj)
+    hubspot_obj = HubspotObject.objects.filter(
+        object_id=obj.id, content_type=content_type
+    ).first()
+    if hubspot_obj:
+        return hubspot_obj.hubspot_id
+    model = obj.__class__
```

Comment:
Just curious why you went with `__class__` over `isinstance()` here?

Reply:
PS That was for xpro where a deal or line_item might be one of two different model objects, but that's not the case here.  So I will definitely switch to `isinstance` for bootcamps - would you recommend doing the same for xpro too?

Reply:
Second argument to `isinstance` can be a tuple of types - `isinstance(obj, (B2BOrder, Order))`